### PR TITLE
Issues #287 - PHP 8.1 

### PR DESCRIPTION
**Pull request by maPer77**
Hello claviska
That should solve question #287
I couldn't fully test it, if you have any more places with this whole number problem let me know

### CODE REVIEW

**Comment by maPer77** [comment]
Only consider the last pull request (@maPer77
Issues claviska#287 - PHP 8.1), I sent the old one together by mistake...

File: src/claviska/SimpleImage.php [hunk 1]
```diff
@@ -490,7 +490,7 @@ protected static function imageCopyMergeAlpha($dstIm, $srcIm, $dstX, $dstY, $src
       imagefilter($srcIm, IMG_FILTER_COLORIZE, 0, 0, 0, 127 * ((100 - $pct) / 100));
     }
 
-    imagecopy($dstIm, $srcIm, $dstX, $dstY, $srcX, $srcY, $srcW, $srcH);
+    imagecopy($dstIm, $srcIm, (int)$dstX, (int)$dstY, (int)$srcX, (int)$srcY, (int)$srcW, (int)$srcH);
```
Comment (claviska) [inline src/claviska/SimpleImage.php:493]:
```suggestion
    imagecopy($dstIm, $srcIm, (int) $dstX, (int) $dstY, (int) $srcX, (int) $srcY, (int) $srcW, (int) $srcH);
```

File: src/claviska/SimpleImage.php [hunk 2]
```diff
@@ -593,7 +593,7 @@ public function crop($x1, $y1, $x2, $y2) {
     // Avoid using native imagecrop() because of a bug with PNG transparency
     $dstW = abs($x2 - $x1);
     $dstH = abs($y2 - $y1);
-    $newImage = imagecreatetruecolor($dstW, $dstH);
+    $newImage = imagecreatetruecolor( (int) $dstW, (int) $dstH );
```
Comment (claviska) [inline src/claviska/SimpleImage.php:596]:
```suggestion
    $newImage = imagecreatetruecolor((int) $dstW, (int) $dstH);
```

File: src/claviska/SimpleImage.php [hunk 3]
```diff
@@ -802,16 +802,16 @@ public function resize($width = null, $height = null) {
     // We can't use imagescale because it doesn't seem to preserve transparency properly. The
     // workaround is to create a new truecolor image, allocate a transparent color, and copy the
     // image over to it using imagecopyresampled.
-    $newImage = imagecreatetruecolor($width, $height);
+    $newImage = imagecreatetruecolor( (int) $width, (int) $height );
```
Comment (claviska) [inline src/claviska/SimpleImage.php:805]:
```suggestion
    $newImage = imagecreatetruecolor((int) $width, (int) $height);
```

File: src/claviska/SimpleImage.php [hunk 4]
```diff
@@ -1885,7 +1885,7 @@ protected function allocateColor($color) {
       $color['red'],
       $color['green'],
       $color['blue'],
-      127 - ($color['alpha'] * 127)
+      (int) ( 127 - ($color['alpha'] * 127) )
```
Comment (claviska) [inline src/claviska/SimpleImage.php:1888]:
```suggestion
      (int) (127 - ($color['alpha'] * 127))
```

File: src/claviska/SimpleImage.php [hunk 5]
```diff
@@ -194,7 +194,7 @@ public function fromFile($file) {
    * @return \claviska\SimpleImage
    */
   public function fromNew($width, $height, $color = 'transparent') {
-    $this->image = imagecreatetruecolor($width, $height);
+    $this->image = imagecreatetruecolor( (int) $width, (int) $height );
```
Comment (claviska) [inline src/claviska/SimpleImage.php:197]:
```suggestion
    $this->image = imagecreatetruecolor((int) $width, (int) $height);
```

File: src/claviska/SimpleImage.php [hunk 6]
```diff
@@ -147,7 +147,7 @@ public function fromFile($file) {
         // workaround to prevent imagepalettetruecolor() from borking transparency.
         $width = imagesx($gif);
         $height = imagesy($gif);
-        $this->image = imagecreatetruecolor($width, $height);
+        $this->image = imagecreatetruecolor( (int) $width, (int) $height );
```
Comment (claviska) [inline src/claviska/SimpleImage.php:150]:
```suggestion
        $this->image = imagecreatetruecolor((int) $width, (int) $height);
```

**Review by claviska — APPROVED** [review]

**Pull request merged**